apps/builder/app/shared/tree-utils.ts

```diff
@@ -493,24 +495,58 @@ export const insertStyleSourceSelectionsCopyMutable = (
 export const insertStylesCopyMutable = (
   styles: Styles,
   copiedStyles: StyleDecl[],
-  copiedStyleSourceIds: Map<StyleSource["id"], StyleSource["id"]>
+  copiedStyleSourceIds: Map<StyleSource["id"], StyleSource["id"]>,
+  mergedBreakpointIds: Map<Breakpoint["id"], Breakpoint["id"]>
 ) => {
   for (const styleDecl of copiedStyles) {
     const newStyleSourceId = copiedStyleSourceIds.get(styleDecl.styleSourceId);
+    // fallback to old id in case breakpoint was added without changes
+    const newBreakpointId =
+      mergedBreakpointIds.get(styleDecl.breakpointId) ?? styleDecl.breakpointId;
     // insert without changes when style source does not have new id
     if (newStyleSourceId === undefined) {
-      const styleDeclKey = getStyleDeclKey(styleDecl);
+      const newStyleDecl = {
+        ...styleDecl,
+        breakpointId: newBreakpointId,
+      };
+      const styleDeclKey = getStyleDeclKey(newStyleDecl);
       // prevent overriding shared styles if already exist
       if (styles.has(styleDeclKey) === false) {
-        styles.set(styleDeclKey, styleDecl);
+        styles.set(styleDeclKey, newStyleDecl);
       }
       continue;
     }
 
     const styleDeclCopy = {
       ...styleDecl,
       styleSourceId: newStyleSourceId,
+      breakpointId: newBreakpointId,
     };
     styles.set(getStyleDeclKey(styleDeclCopy), styleDeclCopy);
   }
 };
+
+const areBreakpointsSimilar = (left: Breakpoint, right: Breakpoint) => {
```

Comment:
this could be part of the css-engine I think, similar to matchMedia, maybe call it equalMedia()?